isobus/src/isobus_virtual_terminal_client.cpp

```diff
@@ -260,98 +265,72 @@ namespace isobus
-		                                                      partnerControlFunction,
-		                                                      CANIdentifier::PriorityLowest7);
+		const std::vector<std::uint8_t> buffer = { static_cast<std::uint8_t>(Function::HideShowObjectCommand),
+			                                         static_cast<std::uint8_t>(objectID & 0xFF),
+			                                         static_cast<std::uint8_t>(objectID >> 8),
+			                                         static_cast<std::uint8_t>(command),
+			                                         0xFF,
+			                                         0xFF,
+			                                         0xFF,
+			                                         0xFF };
+		return queue_command(buffer, CANIdentifier::CANPriority::PriorityLowest7);
 	}
 
-	bool VirtualTerminalClient::send_enable_disable_object(std::uint16_t objectID, EnableDisableObjectCommand command) const
+	bool VirtualTerminalClient::send_enable_disable_object(std::uint16_t objectID, EnableDisableObjectCommand command)
 	{
-		const std::array<std::uint8_t, CAN_DATA_LENGTH> buffer = { static_cast<std::uint8_t>(Function::EnableDisableObjectCommand),
-			                                                         static_cast<std::uint8_t>(objectID & 0xFF),
-			                                                         static_cast<std::uint8_t>(objectID >> 8),
-			                                                         static_cast<std::uint8_t>(command),
-			                                                         0xFF,
-			                                                         0xFF,
-			                                                         0xFF,
-			                                                         0xFF };
-		return CANNetworkManager::CANNetwork.send_can_message(static_cast<std::uint32_t>(CANLibParameterGroupNumber::ECUtoVirtualTerminal),
-		                                                      buffer.data(),
-		                                                      CAN_DATA_LENGTH,
-		                                                      myControlFunction,
-		                                                      partnerControlFunction,
-		                                                      CANIdentifier::PriorityLowest7);
+		const std::vector<std::uint8_t> buffer = { static_cast<std::uint8_t>(Function::EnableDisableObjectCommand),
+			                                         static_cast<std::uint8_t>(objectID & 0xFF),
+			                                         static_cast<std::uint8_t>(objectID >> 8),
+			                                         static_cast<std::uint8_t>(command),
+			                                         0xFF,
+			                                         0xFF,
+			                                         0xFF,
+			                                         0xFF };
+		return queue_command(buffer, CANIdentifier::CANPriority::PriorityLowest7);
```

Comment:
Good question, there are three ways sending a message of less or equal to 8 bytes can fail:
- I added a check that commands only get send when the VT is connected, so messages sent before that fail.
- Address of internal control function get's stolen, so address is invalid and send_message fails
- Invalid state of hardware interface

All three states are pretty unlikely, but since queue_command basically tries to sent it first, we don't get to the queue in 99.9% of the cases. Not sure if the other 0.1% want to be queued? I think there is no performance concerns here, the one extra function in between is negligible

Reply:
OK, yep I agree with your thinking, it's probably ok.